### ISSUE TITLE
🐛Bug: Nested Popover displays incorrectly (happens on doc page)

### ISSUE THREAD

**Issue by duthanhduoc** [melt-ui/melt-ui#367]
**Describe the bug**
Nested Popover displays incorrectly (happens on doc page)

**To Reproduce**
1. Go to [https://www.melt-ui.com/docs/builders/popover](https://www.melt-ui.com/docs/builders/popover)
2. Click Open Nested then click outside then click open nested

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Mac OS 13.4
 - Browser Chrome
 - Version 115.0.5790.114

https://github.com/melt-ui/melt-ui/assets/33114640/94234eb9-796d-4997-9602-0dff095b3dcc

**Comment by salisshuaibu11** [comment]
I've noticed yet another issue: when you click outside the first popover, it closes both the inner popover and the main one, which is good. However, if you click the close icon in the first popover, it only closes the main one and leaves the inner popover open. you will see it when open the first popover, the first will appear along with the second one already opened.

**Comment by huntabyte** [comment]
Thanks for creating the issue! We'll look into this asap!

**Comment by lolcabanon** [comment]
Adding up on this :

Container popover : the original one.
Nested popover: the second one nested inside the Container.

Open Container, Open Nested, Close Nested with "X" : :+1: 

Open Container, Open Nested, Click inside Container (Close Nested), Open Nested : :+1: 

Open Container, Close Container ("X" or click outside), Open Container, Open Nested : :+1: 

Open Container, Open Nested, Close Container (any method or order i.e. close Nested then Container or close both in "1 action", Open Container, Open Nested : :-1: Nested seems to lose track on Container?

As long as the Nested popover was never opened before or opened but the Container popover was not closed, everything works as expected. When the Nested has been opened and the container closed, it stops working.

**Comment by lolcabanon** [comment]
Also at first I was looking through issues because I noticed if there are many popover (not nested) in a page, clicking on another trigger will not close other previously opened popovers even if click outside is enabled.

Maybe these issues are related? Maybe I should open a specific issue?

**Comment by huntabyte** [comment]
> Also at first I was looking through issues because I noticed if there are many popover (not nested) in a page, clicking on another trigger will not close other previously opened popovers even if click outside is enabled.
> 
> Maybe these issues are related? Maybe I should open a specific issue?

If you open an issue with a reproduction we should be able to resolve it asap @lolcabanon

**Comment by lolcabanon** [comment]
> > Also at first I was looking through issues because I noticed if there are many popover (not nested) in a page, clicking on another trigger will not close other previously opened popovers even if click outside is enabled.
> > 
> > Maybe these issues are related? Maybe I should open a specific issue?
> 
> If you open an issue with a reproduction we should be able to resolve it asap @lolcabanon

Cool will work on it today! Seeya ✌️

**Comment by lolcabanon** [comment]
@huntabyte

> If you open an issue with a reproduction we should be able to resolve it asap @lolcabanon

Ok, I tried to replicate in a fresh project, but cannot figure out how to replicate it from scratch.

I updated `@melt-ui/svelte` and `@melt-ui/pp` in my current project and the "bug" is still happening. I tried to simplify the code as much as I can do, but there is still a lot of stuff... I don't think it would be useful to provide you my project code...?

If I stumble on anything helpful i'll let you know!
